pkg/middleware/auth.go

```diff
@@ -188,12 +280,29 @@ func (a Auth) AuthorizeAPI(next http.Handler) http.Handler {
 }
 
 // NewAuth makes Auth middleware
-func NewAuth(authConfig *config.ServiceAuthConfig) Auth {
+func NewAuth(authConfig *config.ServiceAuthConfig) *Auth {
+
+	if authConfig.JwksURL != "" && len(authConfig.HMACSecrets) != 0 {
+		log.Warn().Msg("HMAC Secret will be ignored, JWKS URL will be used for token validation")
+	}
+
+	if authConfig.JwksURL != "" {
+		if authConfig.JwksUpdateInterval <= 0 {
+			log.Error().Msg("JwksUpdateInterval must be set")
+			return nil
+		}
+		verifier := NewJWTVerifierURL(authConfig.JwksURL, authConfig.JwksUpdateInterval)
+		if verifier == nil {
+			log.Error().Msg("problem with getting initial JWKS key set")
```

Comment:
This is leaking a bit as we're inferring why the verifier returned `nil`. The best we can say here is:
```
log.Error().Msg("unable to construct NewJWTVerifierURL")
```